vite.config.js

```diff
@@ -3,54 +3,67 @@ import { defineConfig } from "vite";
 import react from "@vitejs/plugin-react";
 import dotenv from "dotenv";
 import eslintPlugin from "vite-plugin-eslint";
+import { resolve } from "path";
+import { readFileSync } from "fs";
 
 dotenv.config();
 
-export default defineConfig({
-  base: "",
-  root: "",
-  envPrefix: "RECORD_MANAGER_",
-  plugins: [
-    react(),
-    eslintPlugin({
-      cache: false,
-    }),
-  ],
-  build: {
-    sourcemap: true,
-    emptyOutDir: true,
-    rollupOptions: {
-      onwarn: (warning, defaultHandler) => {
-        // TODO: workaround for dangerous use of eval method that should be solved in SForms library issue https://github.com/kbss-cvut/s-forms/issues/283
-        const isFromKbssCvutPackageWarning =
-          warning.code === "EVAL" && warning.message.includes("node_modules/store/plugins/lib/json2.js");
-        // TODO: Rollup Pure Annotation warning should be resolved by https://github.com/kbss-cvut/s-forms/issues/282
-        const isRollupPureAnnotationWarning =
-          warning.code === "INVALID_ANNOTATION" && warning.message.includes("*#__PURE__*");
-        if (isFromKbssCvutPackageWarning || isRollupPureAnnotationWarning) {
-          return;
-        }
-        defaultHandler(warning);
+function loadEnv(mode) {
+  const envPath = resolve(process.cwd(), `.env.${mode}`);
+  const envContent = readFileSync(envPath, "utf-8");
+  const env = dotenv.parse(envContent);
+  return env;
+}
+
+export default ({ mode }) => {
+  const env = loadEnv(mode);
+
+  return defineConfig({
+    base: env.RECORD_MANAGER_BASENAME,
```

Comment:
I might not understand how it works, but

**this seems to me strange to use**. I thought we had environments set up in https://github.com/kbss-cvut/record-manager-ui/blob/9f4470205d545148968d06872a50711fe6c26666/config/index.js#L33

So I am not sure how this setting is useful for us. I thought we would set Router basename instead.